pkg/urbit/jets/c/rip.c

```diff
@@ -1,6 +1,90 @@
 #include "all.h"
 
-u3_noun u3qc_rip(u3_atom bloq, u3_atom b) {
+/*
+  Get the lowest `n` bits of a word `w` using a bitmask.
+*/
+#define TAKEBITS(n,w) \
+  ((n)==32) ? (w) :   \
+  ((n)==0)  ? 0   :   \
+  ((w) & ((1 << (n)) - 1))
+
+/*
+  Divide, rounding up.
+*/
+#define DIVCEIL(x,y) \
+  (x==0) ? 0 :       \
+  1 + ((x - 1) / y);
+
+/*
+  `ripn` breaks `atom` into a list of blocks, of bit-width `bits`. The
+  resulting list will be least-significant block first.
+
+  XX TODO This only handles cases where the bit-width is <= 32.
```

Comment:
Is this `bloq <= 32` or `bloq <= 5`?

Reply:
This is `bloq <= 5`. What I've done with `+repn`/`+rep` and `+ripn`/`+rip` is simply to preserve the existing capabilities, while updating the API (in what I thought would be a difficult-to-stage manner, but that's not really the case now). I intend to rewrite these jets with full argument coverage ASAP.

Reply:
ohh, this is the now-inaccurately named `bits` <= 32, not the actual `bloq` parameter, `rip 9`/(now `run 9` not yet independently jetted) will still work. That seems fine if not ideal yup.